network/network.go

```diff
@@ -276,6 +276,12 @@ func NewNetwork(
 		IPSigner:             peer.NewIPSigner(config.MyIPPort, config.TLSKey),
 	}
 
+	// Invariant: We delay the activation of durango during the TLS handshake to
+	// avoid gossiping any TLS certs that anyone else in the network may
+	// consider invalid. Recall that if a peer gossips an invalid cert, the
+	// connection is terminated.
+	durangoTime := version.GetDurangoTime(config.NetworkID)
```

Comment:
Do we need to iterate over all certs we would consider gossiping and drop any we no longer consider valid? I think the only place we check validity is when we get a cert for the first time?

Reply:
The durango upgrade only makes things more permissive. All certs that were previously valid should still be valid.